src/xtgeo/xyz/points.py

```diff
@@ -55,68 +230,297 @@ def points_from_roxar(
         mypoints = xtgeo.points_from_roxar(project, 'TopEtive', 'DP_seismic')
 
     """
-    obj = Points()
 
-    obj.from_roxar(
-        project,
-        name,
-        category,
-        stype=stype,
-        realisation=realisation,
-        attributes=attributes,
+    return Points(
+        **_roxar_importer(
+            project,
+            name,
+            category,
+            stype,
+            realisation,
+            attributes,
+        )
     )
 
-    return obj
+
+def points_from_surface(
+    regular_surface,
+    zname: str = "Z_TVDSS",
+):
+    """This makes an instance of a Points directly from a RegularSurface object.
+
+    Each surface node will be stored as a X Y Z point.
+
+    Args:
+        regular_surface: XTGeo RegularSurface() instance
+        zname: Name of third column
+
+    .. versionadded:: 2.16
+       Replaces the from_surface() method.
+    """
+
+    return Points(**_surface_importer(regular_surface, zname=zname))
 
 
-class Points(XYZ):  # pylint: disable=too-many-public-methods
-    """Points: Class for a points set in the XTGeo framework.
+def points_from_wells(
+    wells: List[xtgeo.Well],
+    tops: bool = True,
+    incl_limit: Optional[float] = None,
+    top_prefix: str = "Top",
+    zonelist: Optional[list] = None,
+    use_undef: bool = False,
+):
+
+    """Get tops or zone points data from a list of wells.
 
-    The Points class is a subclass of the :class:`.XYZ` class,
-    and the point set itself is a `pandas <http://pandas.pydata.org>`_
+    Args:
+        wells: List of XTGeo well objects.
+            If a list of well files, the routine will try to load well based on file
+            signature and/or extension, but only default settings are applied. Hence
+            this is less flexible and more fragile.
+        tops: Get the tops if True (default), otherwise zone.
+        incl_limit: Inclination limit for zones (thickness points)
+        top_prefix: Prefix used for Tops.
+        zonelist: Which zone numbers to apply, None means all.
+        use_undef: If True, then transition from UNDEF is also used.
+
+    Returns:
+        None if empty data, otherwise a Points() instance.
+
+    Example::
+
+            wells = [xtgeo.Well("w1.w"), xtgeo.Well("w2.w")]
+            points = xtgeo.points_from_wells(wells)
+
+    Note:
+        The deprecated method :py:meth:`~Points.from_wells` returns the number of
+        wells that contribute with points. This is now implemented through the
+        function `get_nwells()`. Hence the following code::
+
+            nwells_applied = poi.from_wells(...)  # deprecated method
+            # vs
+            poi = xtgeo.points_from_wells(...)
+            nwells_applied = poi.get_nwells()
+
+    .. versionadded: 2.16
+
```

Comment:
Add comment:
Replaces the `from_wells()` method

Reply:
Fixed.